pyvirtualcam/native_macos/main.mm

```diff
@@ -0,0 +1,165 @@
+// Rest of files is stolen from:
+// - https://github.com/obsproject/obs-studio/tree/516ed4458dbc74b27b425533789d4e9dd95b923b/plugins/mac-virtualcam
+// - https://github.com/descampsa/yuv2rgb/tree/c51a543cab4322598047bcd86f5d5fd680e9e6f2
+// Consider migration to https://github.com/johnboiles/coremediaio-dal-minimal-example ?
+
+#include <stdexcept>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include "OBSDALMachServer.h"
+#include "Defines.h"
+#include <mach/mach_time.h>
+#include <cstdint>
+
+namespace py = pybind11;
+
+#define UNUSED_PARAMETER(arg)
+
+
+static OBSDALMachServer *sMachServer = nil;
+
+static int cam_output_width;
+static int cam_output_height;
+static uint32_t cam_fps_num;
+static uint32_t cam_fps_den;
+
+
+static void YfromRGB(uint8_t* y, uint8_t r, uint8_t g, uint8_t b) {
+    *y = (uint8_t)( 0.257 * r + 0.504 * g + 0.098 * b +  16);
+}
+static void UVfromRGB(uint8_t* u, uint8_t* v, uint8_t r, uint8_t g, uint8_t b) {
+    *u = (uint8_t)(-0.148 * r - 0.291 * g + 0.439 * b + 128);
+    *v = (uint8_t)( 0.439 * r - 0.368 * g - 0.071 * b + 128);
+}
+
+static bool virtual_output_start(int width, int height, double fps, int delay) {
+    if (sMachServer != nil) {
+        fprintf(stderr, "virtual camera output already started\n");
+        return false;
+    }
+
+    cam_output_width = width;
+    cam_output_height = height;
+    cam_fps_num = fps * 1000;
+    cam_fps_den = 1000;
```

Comment:
Calculation of FPS values is pretty bad. I'm not even sure if I tested this version of the code. Originally I had `fps:1`